joomla/joomla-cms
# Fixed boxchecked value set incorrect.

## Problem

In the list forms where the hidden input value boxchecked is used, it has a default value of 0, no checkboxes are checked. Once you check a box the value of this field is changed to 01 instead of 1. When you uncheck the box, the value becomes 01-1 instead of 2 and it goes on from there.
## Solution

The value must be cast to an integer for JavaScript to be able to calculate with it.
## Testing

To test this issue, you must use the Debug console of the browser to check the initial value of the boxchecked field.
1. Go to the article list
2. Check the initial value of the boxchecked field is 0
3. Click on a checkbox to select it
4. The value becomes 01
5. Uncheck the box and the value becomes 01-1
6. Apply the patch
7. Reload the page
8. Check the initial value of the boxchecked field is 0
9. Click on a checkbox to select it
10. The value becomes 1
11. Uncheck the box and the value becomes 0
12. Check and uncheck more boxes to see the number increases and decreases
